File: 013/removeCharacters.js
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/13/
*/

var fs = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line !== "") {
        line = line.replace(/\r/gm, ""); // For Windows.
        var arr = line.split(", "); // Splits the string by space
        var letters = arr[1].split("");
        var word = arr[0];
        for (var i = 0; i < letters.length; i++) {
            var c = letters[i].toString();
            word = word.split(c);
            word = word.toString();
        }
        word = word.replace(/,/g, "");
        console.log(word);
    }
});

File: 013/removeCharacters.java
/*
	@author Sarah Kidd (www.linkedin.com/in/sarahkidd/)
	Description of problem at: https://www.codeeval.com/open_challenges/13/
*/

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class removeCharacters {
	public static void main(String[] args) {
		try {
			File file = new File(args[0]);
			BufferedReader infile = new BufferedReader(new FileReader(file));
			String line;
			while ((line = infile.readLine()) != null) {
				String[] words = line.split(", ");
				String temp = words[0];
				String[] chars = words[1].split("");
				for (int i = 1; i < chars.length; i++) {
					String c = chars[i] + "*";
					Pattern p = Pattern.compile(c);
					Matcher m = p.matcher(temp);
					temp = m.replaceAll("");
				}
				System.out.println(temp);
			}
			infile.close();
		} catch (FileNotFoundException e) {
			System.out.println("This is crap. Wrong stupid file name.");
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}